### PR TITLE
Fix gnome-shell-extensions-3.28.0

### LINKED ISSUE #38377: gnome-shell-extensions-3.28 are broken on unstable
## Issue description

The meson build doesn't compile the schemas, opting to do it instead in a package shell script. Unfortunately it seems that compiling the schemas aren't enough to make all the extensions work. In particular at least `user-theme` and `apps-menu` still won't work (seems to fix `window-list` though).

With the schemas installed Tweaks simply complains that the `user-theme` extension is incorrectly installed.

`apps-menu` complains about lacking any typelib file for `GMenu`.

I'm not sure how well these extensions worked before 3.28, but a report on irc indicated that `user-theme` at least worked before.

cc @jtojnar 

### Steps to reproduce

Run Tweaks on nixos-unstable, try to enable eg. the `User Themes` extension under `Extensions` or `Applications Menu`. 

## Technical details

Running a schema fix on top of 80e730c9a41ee327308144ffcde47f03a148e7c2

 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.14, NixOS, 18.09.git.e1635e06580 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0`
 - channels(hed): `"unstable-18.03pre126246.a4f433c03cc"`
 - channels(root): `"nixos-17.09.2498.a9ffb7b0b5a"`
 - nixpkgs: `/home/hed/nixpkgs/channel`

### PR DESCRIPTION
**Pull request by hedning**
###### Motivation for this change

This fixes three problems that surfaced when upgrading to gnome 3.28

1. The new meson build doesn't compile the schema files.

2. The `apps-menu` extension broke by `gnome-shell` no longer depending on `gnome-menus`.

3. Lastly `gnome-tweaks` didn't find the necessary files to let users change themes when the `user-theme` extension was enabled.

cc @jtojnar 

fixes #38377 

###### Things done

Light testing of the changes in a VM on top of `nixos-unstable`.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/desktops/gnome-3/core/gnome-shell-extensions/default.nix [hunk 1]
```diff
@@ -16,13 +17,42 @@ stdenv.mkDerivation rec {
...
+    glib-compile-schemas $schemadir
+
+    for f in $extensiondir/*; do
+      name=`basename ''${f%%@*}`
+      uuid=$name@gnome-shell-extensions.gcampax.github.com
+      schema=$schemadir/org.gnome.shell.extensions.$name.gschema.xml
+
+      if [ -f $schema ]; then
+        mkdir $f/schemas
+        cp $schema $f/schemas;
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/core/gnome-shell-extensions/default.nix]: Symlink should be enough.

File: pkgs/desktops/gnome-3/misc/gnome-tweaks/default.nix [hunk 2]
```diff
@@ -23,7 +23,7 @@ in stdenv.mkDerivation rec {
     libnotify gnome3.gnome-shell python3Packages.pygobject3
     libsoup gnome3.gnome-settings-daemon gnome3.nautilus
     gnome3.mutter gnome3.gnome-desktop gobjectIntrospection
-    gnome3.nautilus
+    gnome3.nautilus gnome3.gnome-shell-extensions
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/misc/gnome-tweaks/default.nix]: Could you add comment that it is needed for schemas from the theme extension?

Comment (hedning) [reply pkgs/desktops/gnome-3/misc/gnome-tweaks/default.nix]: Ah, true

File: pkgs/desktops/gnome-3/core/gnome-shell-extensions/default.nix (hunk 1 above)
Comment (hedning) [reply pkgs/desktops/gnome-3/core/gnome-shell-extensions/default.nix]: True

**Review by jtojnar — APPROVED** [review]

**Pull request merged**